src/project.cc

```diff
@@ -35,6 +35,28 @@ void Project::save_files(const boost::filesystem::path &path) {
   }
 }
 
+void Project::on_save() {
+  if(Notebook::get().get_current_page()==-1)
+    return;
+  auto view=Notebook::get().get_current_view();
+  if(view->file_path.filename()=="CMakeLists.txt") {
+    auto build=get_build(view->file_path);
```

Comment:
I think that should be doable as well, by looking for the first CMakeLists.txt in the parent directories and take it from there. I'll add it to my TODO list:)